dlt/common/runners/pool_runner.py

```diff
@@ -59,7 +85,7 @@ def _run_func() -> bool:
         if pool:
             logger.info("Closing processing pool")
             # terminate pool and do not join
-            pool.terminate()
+            pool.shutdown(wait=True)
```

Comment:
I think the old one used a semaphore under the hood in the stdlib. And if the system sigkilled a child process, perhaps due to memory, or a process failed, it could lock. On kubernetes where you have limited CPU, os.cpucount returns incorrect value. Over provisioning procs would stall out the program too.